packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -832,111 +832,111 @@ export function TransferPanel() {
     ]
   )
 
-  const disableDeposit = useMemo(() => {
-    if (
-      isDepositMode &&
-      selectedToken &&
-      isWithdrawOnlyToken(selectedToken.address, l2Network.id)
-    ) {
+  const disableDepositAndWithdrawal = useMemo(() => {
+    if (!amountNum) return true
+    if (isTransferring) return true
+    if (isSwitchingL2Chain) return true
+    if (destinationAddressError) return true
+
+    if (isSmartContractWallet && !selectedToken) {
       return true
     }
 
-    return (
-      isTransferring ||
-      !amountNum ||
-      (isDepositMode &&
-        !isBridgingANewStandardToken &&
-        (!amountNum || !l1Balance || amountNum > +l1Balance)) ||
-      // allow 0-amount deposits when bridging new token
-      (isDepositMode &&
-        isBridgingANewStandardToken &&
-        (l1Balance === null || amountNum > +l1Balance)) ||
-      (isSmartContractWallet && !selectedToken) ||
-      destinationAddressError
-    )
+    // Keep the button disabled while loading gas summary
+    if (!ethBalance || gasSummary.status !== 'success') {
+      return true
+    }
+
+    return false
   }, [
-    isTransferring,
-    isDepositMode,
-    l2Network,
     amountNum,
-    l1Balance,
-    isBridgingANewStandardToken,
-    selectedToken,
+    destinationAddressError,
     isSmartContractWallet,
-    destinationAddressError
+    ethBalance,
+    requiredGasFees,
+    gasSummary.status,
+    isSwitchingL2Chain,
+    isTransferring,
+    selectedToken
   ])
 
-  // TODO: Refactor this and the property above
-  const disableDepositV2 = useMemo(() => {
-    // Keep the button disabled while loading gas summary
-    if (!ethBalance || disableDeposit || gasSummary.status !== 'success') {
+  const disableDeposit = useMemo(() => {
+    if (disableDepositAndWithdrawal) {
       return true
     }
 
+    if (
+      selectedToken &&
+      isWithdrawOnlyToken(selectedToken.address, l2Network.id)
+    ) {
+      return true
+    }
+
+    if (isBridgingANewStandardToken) {
+      if (l1Balance === null || amountNum > +l1Balance) {
+        return true
+      }
+    } else {
+      if (!amountNum || !l1Balance || amountNum > +l1Balance) {
+        return true
+      }
+    }
+
     if (selectedToken) {
+      if (!ethBalance) {
```

Comment:
`disableDepositAndWithdrawal` already handled `!ethBalance`
why do we need this extra condition here?